src/python/pants/backend/native/targets/native_library.py

```diff
@@ -0,0 +1,78 @@
+# coding=utf-8
+# Copyright 2018 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import (absolute_import, division, generators, nested_scopes, print_function,
+                        unicode_literals, with_statement)
+
+from pants.backend.native.targets.native_artifact import NativeArtifact
+from pants.base.exceptions import TargetDefinitionException
+from pants.base.payload import Payload
+from pants.base.payload_field import PrimitiveField
+from pants.build_graph.target import Target
+
+
+class NativeLibrary(Target):
+  """A class wrapping targets containing sources for C-family languages and related code."""
+
+  @classmethod
+  def produces_ctypes_dylib(cls, target):
```

Comment:
`ctypes` is python specific, so it's a bit odd for references to it to be here.

Presumably you're trying to avoid needing to create new wrapper target types for each language that might consume the library... if so, it might be worth discouraging the next person who comes along from doing that (and instead encouraging them to create the separate target to have a relationship like `python_library depends on ctype_library dependes on c_library`). Alternatively, consider just doing that here?

Reply:
The reason for adding `ctypes` to any symbol names was so that it's clear when used they are not capable of any other use yet (and also, because of name clashes with `contrib/cpp`). The idea is that there is just `c/cpp_library`, declaring maybe a `produces_library=native_artifact()`, and that can be a dependency for multiple different languages, and the names would be modified as functionality is added.